web/src/components/data-entities/survey/SpeciesCorrectEdit.js

```diff
@@ -19,15 +21,21 @@ import CustomSearchFilterInput from '../../input/CustomSearchFilterInput';
 import {searchSpecies, postSpeciesCorrection} from '../../../api/api';
 import {Paper} from '@mui/material';
 
-import {makeStyles} from '@mui/styles';
+const PREFIX = 'SpeciesCorrectEdit';
+
+const classes = {
+  root: `${PREFIX}-root`
+};
```

Comment:
```javascript
const PREFIX = 'SomeName';
const classes = {
  root: `${PREFIX}-root`
};
```

this block appears the same in multiple places, is there any specific reason it cannot be just

```javascript
const classes = {
  root: `SomeName-root`
};
```
even so, I'd refer to have list of constants grouped in a separate file to import later 

e.g

create a file `constant.js`

```javascript
const SOME_NAME_CLASSES = {
  root: "SomeName-root"
};

const ANOTHER_NAME_CLASSES = {
  root: "AnotherName-root"
};

export {
  SOME_NAME_CLASSES,
  ANOTHER_NAME_CLASSES
};
```

then in the component you need one of the above constant, you can import it with

```javascript
import { SOME_NAME_CLASSES as classes } from "./path/to/constants.js";
```

Reply:
@HavierD

Reply:
having 

```javascript
const PREFIX = ...;
const classes = ...;
```
spread everywhere may not be ideal. please change.

Reply:
Thank you Viet. I will check it later after finishing my current ticket.

Reply:
@vietnguyengit I remember the reason. After researching the usage of `@mui/material/styles`, it must have a unique class name otherwise it will have CSS class collisions. At the beginning, my code is similar with this: 
```javascript
const classes= {
  root: "SomeName-root"
};
```

But some components have multiple style class, i have to be like this: 
```javascript
const classes= {
  root: "SomeName-root",
  header:"SomeName-header",
  body":"SomeName-body"
...
};
```

So I think, extracting the prefix of the classes is a good idea.

And for consistence and maintenance, i extract all the prefixes if the `styled()` from `@mui/material/styles` is in use, even though some of them only have one class. 

To make sure the prefix is unique, i use the component name as the prefix (I know that occasionally the name might be duplicated, but it is easy to solve and for the current nrmn, this is enough.).  
So currently, all the components using `styled()` have this structure (which is neat and easy to understand and maintain I reckon):

```javascript
const PREFIX = <component_name>
const classes = {
  <name1>: `${PREFIX}-name1`,
  <name2>: `${PREFIX}-name2`
 ...
}

```
And because the PREFIX is just the same name with the component, so i think putting PREFIX into the component file rather than putting all PREFIXs into one `constraint.js` is better.

The above is my understanding and explanation. Please let me know if something is incorrect.

Reply:
```javascript
const PREFIX = <component_name>
const classes = {
  <name1>: `${PREFIX}-name1`,
  <name2>: `${PREFIX}-name2`
 ...
}
```

which could looklike

```javascript
const SOME_NAME_CLASSES = {
  root: "SomeName-root"
};

const ANOTHER_NAME_CLASSES = {
  root: "AnotherName-root"
};

const NEW_CLASSES = {
  <name1>: "Component-name1",
  <name2>: "Component-name2"
 ...
}

export {
  SOME_NAME_CLASSES,
  ANOTHER_NAME_CLASSES,
  NEW_CLASSES
};
```

you still can declare the constants from dedicated/centralised place for all the constants, my point is you introduced duplicate code structure everywhere unnecessarily.

Please make new PR to maintain good code practice. @HavierD

Reply:
@HavierD  https://semaphoreci.com/blog/constants-layer-javascript#:~:text=such%20a%20layer!-,3%20Approaches%20for%20Organizing%20Constants%20in%20a%20JavaScript%20Project,-Let%E2%80%99s%20see%20three

Reply:
different language but still the same point : https://medium.com/swlh/why-you-should-use-a-constants-file-in-swift-ff8c40af1b39

Reply:
https://dev.to/amirfakour/tips-to-use-constants-file-in-typescript-27je

Reply:
I know your point. If several files share one PREFIX & classes, it is a good idea to put them together into a constant.js (like some normal constants such as names, URLs, theme parameters etc.). But the PREFIXes i am using are privately used. They can only be used in one file, and must not be shared (will cause css class collision). Putting them together will make them be public rather than private. As we all know, make private members public is dangerous.

Reply:
> I know your point. If several files share one PREFIX & classes, it is a good idea to put them together into a constant.js (like some normal constants such as names, URLs, theme parameters etc.). But the PREFIXes i am using are privately used. They can only be used in one file, and must not be shared (will cause css class collision). Putting them together will make them be public rather than private. As we all know, make private members public is dangerous.

if you are sure they wont be used anywhere else, and like you mentioned that the whole point you tried to do it is for having unique names as required by your new UI dependency, then again, you still litter the code with duplicate structure everywhere

```javascript
const PREFIX = <component_name>
const classes = {
  <name1>: `${PREFIX}-name1`,
  <name2>: `${PREFIX}-name2`
 ...
}
```

please make new PR to avoid using duplicate duplicate and hard-coded `PREFIX` approach, you can use **dynamic** & unique string approach for example, because like you said, nowhere else uses them,  @HavierD

Reply:
even you don't use the exported const anywhere, they are still constants 

>  make private members public is dangerous.

i dont think you understand my feedback and the sample articles? @HavierD

Reply:
again, please make new PRs to improve your code standard to avoid using duplicate code structure unnecessarily. I moved the ticket to `In progress` and put it as a blocker until you make improvements. @HavierD